juju/juju
# audit-log: Handle changes to audit configuration in the API server

## Description of change

The changes in #8315 allowed updating the controller config for audit logging. This PR changes the machine agent and API server to watch controller config for changes in the audit logging settings, and use them in new API connections.

It disallows changing `audit-log-max-backups` and `-max-size` - handling these properly would require some more architectural changes to make sure that the log file gets closed and re-opened correctly for requests that are in progress.

## QA steps

Bootstrap a controller with `auditing-enabled=false`.
Deploy some applications and add relations and check that API requests aren't written to the audit log.
Run `juju controller-config auditing-enabled=true`.
Make some more changes to the model (add/remove units and relations).
See that those changes are now being written to the audit log.
Change the values of `audit-log-exclude-methods` and `audit-log-capture-args` - check that the changes are reflected in the records written out.
Turn audit logging off, and make more changes.
Turn it on again, and make more changes.
